File: modules/wem-api/src/main/java/com/enonic/wem/api/util/BinaryReference.java
package com.enonic.wem.api.util;

public class BinaryReference
{
    private final String value;

    private BinaryReference( final String value )
    {
        this.value = value;
    }

    public static BinaryReference from( final String value )
    {
        return new BinaryReference( value );
    }

    @Override
    public String toString()
    {
        return this.value;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }

        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        final BinaryReference that = (BinaryReference) o;

        if ( value != null ? !value.equals( that.value ) : that.value != null )
        {
            return false;
        }

        return true;
    }

    @Override
    public int hashCode()
    {
        return value != null ? value.hashCode() : 0;
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/util/BinaryReference.ts
module api.util {

    export class BinaryReference implements api.Equitable {

        private value: string;

        constructor(value: string) {
            this.value = value;
        }

        getValue(): string {
            return this.value;
        }

        equals(o: Equitable): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, BinaryReference)) {
                return false;
            }

            var other = <BinaryReference>o;

            if (!api.ObjectHelper.stringEquals(this.value, other.value)) {
                return false;
            }

            return true;
        }

        toString(): string {
            return this.value;
        }
    }
}